### ISSUE TITLE
Normalization cannot work

### ISSUE THREAD

**Issue by gucascau** [satijalab/seurat#7206]
Hi team,

I create an object with this matrix input, but when I run with normalization. I kept got the errors

> UUOProject[1:3,1:3]

  P5877_GTCGGAGTTCCGCAAGTTC P5539_CAAGGCGTTAACTGCTGCC P5932_TGCGTAGTTGTTCGTCAGA
  0610007P14Rik                         0                         0                         0
  0610009B22Rik                         0                         0                         0
  0610009L18Rik                         0                         0                         0

> UUOProjectObject<- CreateSeuratObject(counts = UUOProject, project= "Urotherlium",assay = "RNA")

> UUOProjectObject <- NormalizeData(UUOProjectObject)

Error in GetAssay.Seurat(object = object, assay = assay) : 
  RNA is not an assay present in the given object. Available assays are: 

Not sure what would be the reason. I tried to setup assay to RNA but still not work.

When I used the command line below, it seems to work very well. However, it will not work for the following FindVariableFeatures, scale.

> UUOProjectObject[["RNA"]]$counts<-NormalizeData(UUOProjectObject[["RNA"]]$counts)

Could you please give some suggestions? Many many thanks.

**Comment by YidaZhang0628** [comment]
I have encountered the same issue. Hope they can provide some comment

**Comment by YidaZhang0628** [comment]
Something I noticed is that if I create the Seurat object using Seurat V5 `(options(Seurat.object.assay.version = "v5"))`, I will encounter this problem even when the assay is in the object. But if I don't use Seurat V5, I can run everything without a problem.

**Comment by gucascau** [comment]
Yea, I struggled a lot for the V5. V 3 works well for this normalization.

**Comment by maithermbarros** [comment]
I am having exactly the same problem. 

`<html>
<body>


> Assays(obj_patient1) [1] "RNA" > class(obj_patient1[["RNA"]]) [1] "Assay5" attr(,"package") [1] "SeuratObject" > patient1_within <- NormalizeData(obj_patient1) Error in GetAssay.Seurat(object = object, assay = assay) :    RNA is not an assay present in the given object. Available assays are:
--
 
> | >
>


</body>
</html>`

**Comment by Gesmira** [comment]
Hi, could you include the output of `sessionInfo`? I can reproduce the GetAssay() issue, however, NormalizeData seems to run normally if Seurat v5 is installed.
```
remotes::install_github("satijalab/seurat", "seurat5", quiet = TRUE)
library(Seurat)
```

**Comment by gong-dennis** [comment]
I am having the same issue. Here is my sessionInfo:

`R version 4.2.2 (2022-10-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.4

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.14.8       lubridate_1.9.2         forcats_1.0.0          
 [4] stringr_1.5.0           dplyr_1.1.2             purrr_1.0.1            
 [7] readr_2.1.4             tidyr_1.3.0             tibble_3.2.1           
[10] ggplot2_3.4.2           tidyverse_2.0.0         Seurat_4.9.9.9044      
[13] SeuratObject_4.9.9.9084 sp_1.6-0               

loaded via a namespace (and not attached):
  [1] Rtsne_0.16             colorspace_2.1-0       deldir_1.0-6          
  [4] ellipsis_0.3.2         ggridges_0.5.4         rprojroot_2.0.3       
  [7] rstudioapi_0.14        spatstat.data_3.0-1    leiden_0.4.3          
 [10] listenv_0.9.0          remotes_2.4.2          ggrepel_0.9.3         
 [13] fansi_1.0.4            codetools_0.2-19       splines_4.2.2         
 [16] polyclip_1.10-4        spam_2.9-1             jsonlite_1.8.4        
 [19] ica_1.0-3              cluster_2.1.4          png_0.1-8             
 [22] uwot_0.1.14            shiny_1.7.4            sctransform_0.3.5     
 [25] spatstat.sparse_3.0-1  compiler_4.2.2         httr_1.4.5            
 [28] Matrix_1.5-3           fastmap_1.1.1          lazyeval_0.2.2        
 [31] cli_3.6.1              later_1.3.1            prettyunits_1.1.1     
 [34] htmltools_0.5.5        tools_4.2.2            igraph_1.4.2          
 [37] dotCall64_1.0-2        gtable_0.3.3           glue_1.6.2            
 [40] RANN_2.6.1             reshape2_1.4.4         Rcpp_1.0.10           
 [43] scattermore_1.0        vctrs_0.6.2            spatstat.explore_3.1-0
 [46] nlme_3.1-162           progressr_0.13.0       lmtest_0.9-40         
 [49] spatstat.random_3.1-4  ps_1.7.4               globals_0.16.2        
 [52] timechange_0.2.0       mime_0.12              miniUI_0.1.1.1        
 [55] lifecycle_1.0.3        irlba_2.3.5.1          goftest_1.2-3         
 [58] future_1.32.0          MASS_7.3-58.3          zoo_1.8-12            
 [61] scales_1.2.1           hms_1.1.3              promises_1.2.0.1      
 [64] spatstat.utils_3.0-2   parallel_4.2.2         RColorBrewer_1.1-3    
 [67] curl_5.0.0             reticulate_1.28        pbapply_1.7-0         
 [70] gridExtra_2.3          stringi_1.7.12         desc_1.4.2            
 [73] pkgbuild_1.4.0         rlang_1.1.1            pkgconfig_2.0.3       
 [76] matrixStats_0.63.0     lattice_0.21-8         ROCR_1.0-11           
 [79] tensor_1.5             patchwork_1.1.2        htmlwidgets_1.6.2     
 [82] processx_3.8.0         cowplot_1.1.1          tidyselect_1.2.0      
 [85] parallelly_1.35.0      RcppAnnoy_0.0.20       plyr_1.8.8            
 [88] magrittr_2.0.3         R6_2.5.1               generics_0.1.3        
 [91] DBI_1.1.3              pillar_1.9.0           withr_2.5.0           
 [94] fitdistrplus_1.1-11    survival_3.5-5         abind_1.4-5           
 [97] future.apply_1.10.0    crayon_1.5.2           KernSmooth_2.23-20    
[100] utf8_1.2.3             spatstat.geom_3.1-0    plotly_4.10.1         
[103] tzdb_0.3.0             grid_4.2.2             callr_3.7.3           
[106] digest_0.6.31          xtable_1.8-4           httpuv_1.6.9          
[109] munsell_0.5.0          viridisLite_0.4.2  `

**Comment by gong-dennis** [comment]
I noticed that after detaching Seurat and reattaching it following installation, normalization runs smoothly

**Comment by Gesmira** [comment]
The GetAssay error should now be fixed in the most recent version of seurat5, so NormalizeData should also function normally. 
You can install with 
`remotes::install_github("satijalab/seurat", "seurat5", quiet = TRUE)`

**Comment by uqnsarke** [comment]
My normalization also not working. Any feedback is highly appreciated.

My code is below:

#create count matrix
cts <- ReadMtx(mtx = '/home/sarkern2/R/test/single_cell_rnaseq/data/raw_feature_bc_matrix/matrix.mtx.gz',
	features = '/home/sarkern2/R/test/single_cell_rnaseq/data/raw_feature_bc_matrix/features.tsv.gz',
	cells = '/home/sarkern2/R/test/single_cell_rnaseq/data/raw_feature_bc_matrix/barcodes.tsv.gz')
cts[1:10,1:10]

#Create Seurat objects
pbmc.seurat <- CreateSeuratObject(counts = cts)
str(pbmc.seurat)

# explore QC
pbmc.seurat$mitoPercent <- PercentageFeatureSet(pbmc.seurat, pattern = '^MT-')

# Filter cells
pbmc.seurat.filtered <- subset(pbmc.seurat, subset = nCount_RNA > 800 & nFeature_RNA > 500 & mitoPercent < 10)

# data pre-processing & normalization
pbmc.seurat.filtered <- NormalizeData(object = pbmc.seurat.filtered)

**Comment by hukai916** [comment]
I encountered the same issue when I accidentlly installed Seurat v5 (seems like v5 is not super stable yet), and I decided to downgrade to v4, however, the `NormalizeData()` function refuse to work as before. And even if I uninstall Seurat v5 and re-install Seurat v4, the `CreateSeuratObject()` function would still create a "Assay5" class object, which seems not compatible with `NormalizeData()`. 

I later found that when I install Seurat v5, the dependent library `SeuratObject` was upgraded to `4.9`, and I need to uninstall `SeuratObject` first and then reinstall Seurat v4 so to get the old SeuratObject v4.1.4 back. After that, `NormalizeData()` works as before.
